sdk/core/src/main/java/com/klaviyo/coresdk/config/KlaviyoConfig.kt

```diff
@@ -14,111 +17,111 @@ class KlaviyoMissingAPIKeyException : Exception("You must declare an API key for
 
-        fun networkFlushInterval(networkFlushInterval: Int) = apply {
+        override fun networkFlushInterval(networkFlushInterval: Int) = apply {
             if (networkFlushInterval < 0) {
                 // TODO: When Timber is installed, log warning here
             } else {
                 this.networkFlushInterval = networkFlushInterval
             }
         }
 
-        fun networkFlushDepth(networkFlushDepth: Int) = apply {
+        override fun networkFlushDepth(networkFlushDepth: Int) = apply {
             if (networkFlushDepth <= 0) {
                 // TODO: When Timber is installed, log warning here
             } else {
                 this.networkFlushDepth = networkFlushDepth
             }
         }
 
-        fun networkFlushCheckInterval(networkFlushCheckInterval: Int) = apply {
-            if (networkFlushCheckInterval < 0) {
-                // TODO: When Timber is installed, log warning here
-            } else {
-                this.networkFlushCheckInterval = networkFlushCheckInterval
-            }
-        }
-
-        fun networkUseAnalyticsBatchQueue(networkUseAnalyticsBatchQueue: Boolean) = apply {
-            this.networkUseAnalyticsBatchQueue = networkUseAnalyticsBatchQueue
-        }
-
-        internal fun clock(clock: Clock) = apply {
-            this.clock = clock
-        }
-
-        fun build() {
+        override fun build(): Config {
             if (apiKey.isEmpty()) {
                 throw KlaviyoMissingAPIKeyException()
             }
             if (applicationContext == null) {
                 throw KlaviyoMissingContextException()
             }
 
+            val permissions = applicationContext!!.packageManager.getPackageInfoCompat(
+                applicationContext!!.packageName, PackageManager.GET_PERMISSIONS
+            ).requestedPermissions ?: arrayOf()
+
+            if (Manifest.permission.ACCESS_NETWORK_STATE !in permissions) {
+                throw KlaviyoMissingPermissionException(Manifest.permission.ACCESS_NETWORK_STATE)
+            }
+
             KlaviyoConfig.apiKey = apiKey
             KlaviyoConfig.applicationContext = applicationContext as Context
             KlaviyoConfig.networkTimeout = networkTimeout
             KlaviyoConfig.networkFlushInterval = networkFlushInterval
             KlaviyoConfig.networkFlushDepth = networkFlushDepth
-            KlaviyoConfig.networkFlushCheckInterval = networkFlushCheckInterval
-            KlaviyoConfig.networkUseAnalyticsBatchQueue = networkUseAnalyticsBatchQueue
             KlaviyoConfig.clock = clock
+
+            return KlaviyoConfig
         }
     }
 }
+
+internal fun PackageManager.getPackageInfoCompat(packageName: String, flags: Int = 0): PackageInfo =
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
+        getPackageInfo(packageName, PackageManager.PackageInfoFlags.of(flags.toLong()))
+    } else {
+        @Suppress("DEPRECATION") getPackageInfo(packageName, flags)
+    }
```

Comment:
https://developer.android.com/reference/androidx/core/content/PackageManagerCompat
i think this exists?

Reply:
I did find that, but I couldn't figure out how to access it properly in this context. I think the getter is only available as an extension function of Activity